packages/@lwc/features/src/babel-plugin/index.js

```diff
@@ -8,24 +8,30 @@ const defaultFeatureFlags = require('../../').default;
 
 const RUNTIME_FLAGS_IDENTIFIER = 'runtimeFlags';
 
-function isRuntimeFlag(path, featureFlags) {
-    return (
-        path.isMemberExpression() &&
-        path.get('object').isIdentifier({ name: RUNTIME_FLAGS_IDENTIFIER }) &&
-        path.get('property').isIdentifier() &&
-        featureFlags[path.node.property.name] !== undefined
-    );
-}
-
-function validate(name, value) {
-    if (!/[A-Z_]+/.test(name)) {
+function validate(memberExpressionPath, { featureFlags, opts }) {
+    const name = memberExpressionPath.node.property.name;
+    const value = featureFlags[name];
+    if (!/^[A-Z_]+$/.test(name)) {
         throw new Error(
             `Invalid feature flag "${name}". Flag name must only be composed of uppercase letters and underscores.`
         );
     }
     if (value === undefined) {
         throw new Error(`Invalid feature flag "${name}". Flag is undefined.`);
     }
+
+    const objectPath = memberExpressionPath.get('object');
+    const isRuntimeLookup = objectPath.isIdentifier({ name: RUNTIME_FLAGS_IDENTIFIER });
+    if (isRuntimeLookup && !opts.prod) {
+        throw new Error(
+            `Runtime flags should never be used directly and should only be added by the compiler.`
+        );
+    }
+}
+
+function isBindingReference(path, scope) {
+    const binding = scope && scope.getBinding(path.node.name);
+    return !!(binding && binding.referencePaths.includes(path));
```

Comment:
does 'binding' have a default value for referencePaths?

Reply:
Yeah, the default value is an empty array.